File: LeetCode/Medium/3315. Construct the Minimum Bitwise Array II.rb
# @param {Integer[]} nums
# @return {Integer[]}
def min_bitwise_array(nums)
    nums.map do |num|
        p = 1

        while p <= num and (num & p) != 0 do
            p <<= 1
        end

        if p > 1
            num ^ (p >> 1)
        else
            -1
        end
    end
end

File: LeetCode/Easy/3315. Construct the Minimum Bitwise Array II.java
class Solution {
    public int[] minBitwiseArray(List<Integer> nums) {
        final int[] answer = new int[nums.size()];

        for (int i = 0, j = 1; i < answer.length; i++, j = 1) {
            final int num = nums.get(i);

            while ((num & j) != 0) {
                j <<= 1;
            }

            if (j == 1) {
                answer[i] = -1;
            } else {
                answer[i] = num ^ (j >> 1);
            }
        }

        return answer;
    }
}